### PR TITLE
Refactor outputs for 0.12, switch from maps to lists

### PR DESCRIPTION
**Pull request by ludoo**
This is one of the PRs we need for Fabric, that switch outputs in modules supporting multiple resources from maps to list. The rationale is to keep the same order passed in input variables (eg `names` here) for outputs, which lists do but maps don't as they reorder keys lexically. Having a different order in output vars tripped up ourselves and a few customers in the past.

This also:

- adds outputs for the base `google_service_account` resources, since it's now possible with Terraform 0.12;
- fixes an issue with `formatlist` in 0.12 where the resulting list always has a nominal length of 1 and cannot e combined with other list of the same (effective) length, switching to using the new `for` construct is cleaner and works better.

It's a backwards-incompatible change, so it needs a major version bump. Tests and example have been updated accordingly.

### CODE REVIEW

**Comment by ludoo** [comment]
Refactored following yesterday's chat discussion: the map-style outputs are preserved, and new list-style outputs have been added. All tests pass (and they too were refactored to test both styles).

File: outputs.tf [hunk 1]
```diff
@@ -41,10 +41,20 @@ output "service_accounts" {
 
 output "emails" {
   description = "Service account emails."
-  value       = google_service_account.service_accounts[*].email
+  value       = zipmap(var.names, google_service_account.service_accounts[*].email)
```
Comment (ingwarr) [inline outputs.tf]: Maybe better ` value       = zipmap(var.names, google_service_account.service_accounts.*.email)`

File: outputs.tf [hunk 2]
```diff
@@ -41,10 +41,20 @@ output "service_accounts" {
...
 }
 
 output "iam_emails" {
+  description = "IAM-format service account emails."
+  value       = zipmap(var.names, local.iam_emails)
+}
+
+output "emails_list" {
+  description = "Service account emails."
+  value       = google_service_account.service_accounts[*].email
```
Comment (ingwarr) [inline outputs.tf]: Same `value       = google_service_account.service_accounts.*.email`

File: outputs.tf (hunk 1 above)
Comment (ludoo) [reply outputs.tf]: I kind of prefer the new splat syntax to the legacy one. What's your motivtion for using legacy?

File: outputs.tf (hunk 2 above)
Comment (ludoo) [reply outputs.tf]: Same as above.

**Comment by ludoo** [comment]
Re ingwarr's suggestions to switch back to the legacy splat operator, this is the [recommended best practice](https://www.hashicorp.com/blog/terraform-0-12-generalized-splat-operator#full-splat-operator) on the Terraform documentation:

```We recommend using the full splat syntax for all new configurations since its behavior is generally more intuitive. The older syntax is retained for backwards compatibility but is not recommended.```

**Comment by ingwarr** [comment]
@ludoo Ok, you are right, thank you

**Comment by ludoo** [comment]
> @ludoo Ok, you are right, thank you

I think you need to approve, since you requested changes ;)

File: test/integration/multiple_service_accounts/controls/gcp.rb [hunk 3]
```diff
@@ -15,15 +15,18 @@
 control "gcp" do
   title "GCP Resources"
 
-  attribute('emails').each do |email|
-    describe google_service_accounts(project: "#{attribute('project_id')}") do
-      its('service_account_emails'){ should include email }
-    end
+  attribute('iam_emails').each_value do |email|
     describe google_project_iam_binding(project: "#{attribute("project_id")}",  role: 'roles/viewer') do
-      its('members') {should include "serviceAccount:#{email}" }
```
Comment (morgante) [inline test/integration/multiple_service_accounts/controls/gcp.rb]: Shouldn't we keep the `serviceAccount` component?

File: outputs.tf [hunk 4]
```diff
@@ -14,52 +14,62 @@
...
 }
 
 output "keys" {
   description = "Map of service account keys."
   sensitive   = true
-  value = zipmap(
-    formatlist("%s-key.json", var.names),
-    data.template_file.keys.*.rendered,
-  )
+  value       = zipmap(var.names, data.template_file.keys[*].rendered)
```
Comment (morgante) [inline outputs.tf]: This looks like a breaking change. We're changing the keys from `{names}-key.json` to `names` directly. Do we think this change is worth incrementing by a major version?

File: test/integration/multiple_service_accounts/controls/gcp.rb (hunk 3 above)
Comment (ludoo) [reply test/integration/multiple_service_accounts/controls/gcp.rb]: We are keeping it, I am simply testing on the different outputs: the new `emails_list` (list format, no `serviceAccount:` interpolation), and `iam_emails` (map format, interpolation in the output itself so no need to do it in the test), while previously we were testing a single output with no interpolation, hence the need to do it in the test. This makes sure both outputs are as expected.

File: outputs.tf (hunk 4 above)
Comment (ludoo) [reply outputs.tf]: I don't think so, as those outputs are a) security sensitive so I don't think many people use them, and b) absolutely not meant to be used as input in other resources. The only thing that could break are scripts that rely on those to extract the keys.

TL;DR: I can revert, or we can bump to a major version if you think the change warrants it. I kind of don't but am fine with either.

Comment (morgante) [reply outputs.tf]: I think we can go ahead and increment by a major version to avoid breaking any scripts. A "major version" just means we have an interface-breaking change.
